src/java.base/share/classes/sun/security/provider/SunEntries.java

```diff
@@ -278,7 +282,12 @@ public final class SunEntries {
          * Set the digest provider based on whether native crypto is
          * enabled or not.
          */
-        if (useNativeDigest) {
+        if (useNativeDigest
+/*[IF CRIU_SUPPORT]*/
+            && !InternalCRIUSupport.isCheckpointAllowed()
```

Comment:
Like in `SunJCE`, I think this disables the native implementation even after a restore.

Reply:
Same as